cmake/modules/SwiftWindowsSupport.cmake

```diff
@@ -1,77 +1,83 @@
+  precondition("$ENV{UCRTVersion}"
+    MESSAGE "UCRTVersion environment variable must be set")
 endfunction()
 
-function(swift_windows_include_for_arch arch var)
+
+function(swift_windows_include_for_arch arch result_var)
   swift_verify_windows_environment_variables()
 
   set(paths
-        "$ENV{VCToolsInstallDir}/include"
-        "$ENV{UniversalCRTSdkDir}/Include/$ENV{UCRTVersion}/ucrt"
-        "$ENV{UniversalCRTSdkDir}/Include/$ENV{UCRTVersion}/shared"
-        "$ENV{UniversalCRTSdkDir}/Include/$ENV{UCRTVersion}/um")
-  set(${var} ${paths} PARENT_SCOPE)
+    "$ENV{VCToolsInstallDir}/include"
+    "$ENV{UniversalCRTSdkDir}/Include/$ENV{UCRTVersion}/ucrt"
+    "$ENV{UniversalCRTSdkDir}/Include/$ENV{UCRTVersion}/shared"
+    "$ENV{UniversalCRTSdkDir}/Include/$ENV{UCRTVersion}/um")
+  set(${result_var} "${paths}" PARENT_SCOPE)
 endfunction()
 
-function(swift_windows_lib_for_arch arch var)
+
+function(swift_windows_lib_for_arch arch result_var)
   swift_verify_windows_environment_variables()
-  swift_windows_arch_spelling(${arch} ARCH)
+  swift_windows_arch_spelling("${arch}" windows_arch)
 
   set(paths)
-  if(${ARCH} STREQUAL x86)
+  if("${windows_arch}" STREQUAL "x86")
     list(APPEND paths "$ENV{VCToolsInstallDir}/Lib")
   else()
-    list(APPEND paths "$ENV{VCToolsInstallDir}/Lib/${ARCH}")
+    list(APPEND paths "$ENV{VCToolsInstallDir}/Lib/${windows_arch}")
   endif()
   list(APPEND paths
-          "$ENV{UniversalCRTSdkDir}/Lib/$ENV{UCRTVersion}/ucrt/${ARCH}"
-          "$ENV{UniversalCRTSdkDir}/Lib/$ENV{UCRTVersion}/um/${ARCH}")
-  set(${var} ${paths} PARENT_SCOPE)
+    "$ENV{UniversalCRTSdkDir}/Lib/$ENV{UCRTVersion}/ucrt/${windows_arch}"
+    "$ENV{UniversalCRTSdkDir}/Lib/$ENV{UCRTVersion}/um/${windows_arch}")
+  set(${result_var} "${paths}" PARENT_SCOPE)
 endfunction()
 
+
 function(swift_windows_generate_sdk_vfs_overlay flags)
   swift_verify_windows_environment_variables()
 
-  get_filename_component(VCToolsInstallDir $ENV{VCToolsInstallDir} ABSOLUTE)
-  get_filename_component(UniversalCRTSdkDir $ENV{UniversalCRTSdkDir} ABSOLUTE)
-  set(UCRTVersion $ENV{UCRTVersion})
+  get_filename_component(VCToolsInstallDir "$ENV{VCToolsInstallDir}" ABSOLUTE)
+  get_filename_component(UniversalCRTSdkDir "$ENV{UniversalCRTSdkDir}" ABSOLUTE)
+  set(UCRTVersion "$ENV{UCRTVersion}")
 
   # TODO(compnerd) use a target to avoid re-creating this file all the time
-  configure_file("${CMAKE_SOURCE_DIR}/utils/WindowsSDKVFSOverlay.yaml.in"
-                 "${CMAKE_BINARY_DIR}/windows-sdk-vfs-overlay.yaml"
-                 @ONLY)
+  configure_file(
+    "${CMAKE_SOURCE_DIR}/utils/WindowsSDKVFSOverlay.yaml.in"
+    "${CMAKE_BINARY_DIR}/windows-sdk-vfs-overlay.yaml"
+    @ONLY)
 
   set(${flags}
-      -Xclang;-ivfsoverlay;-Xclang;"${CMAKE_BINARY_DIR}/windows-sdk-vfs-overlay.yaml"
-      PARENT_SCOPE)
+    "-Xclang" "-ivfsoverlay"
+    "-Xclang" "${CMAKE_BINARY_DIR}/windows-sdk-vfs-overlay.yaml"
```

Comment:
Id really rather not change this part.  I've had CMake "unique" the flags and lose some of them.  Please do a full windows build if you are going to change this.

Reply:
Yeah, it would drop the `-Xclang` sometimes.  I never figured out why/how it was doing that, but that was something that I saw.